### PR TITLE
cmd/trace-agent: enhance DD_APM_FILTER_TAGS_* support

### PR DESCRIPTION
**Pull request by ajgajg1134**
### What does this PR do?

Support array values for these two environment variables. This allows customers to filter or require tag values that have a space in them.

### Motivation

This is based on a customer support request APMS-7131

### Describe how to test/QA your changes

Can be tested by setting the environment variable for the running agent to include this new format
DD_APM_FILTER_TAGS_REQUIRE='["someKey:some Value"]'
DD_APM_FILTER_TAGS_REJECT='["someBadKey:some Bad Value"]'

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by alai97 — APPROVED** [review]

File: cmd/trace-agent/config/config_test.go [hunk 1]
```diff
@@ -841,6 +841,16 @@ func TestLoadEnv(t *testing.T) {
 		assert.NoError(err)
 		assert.Equal(cfg.RequireTags, []*config.Tag{{K: "important1", V: ""}, {K: "important2", V: "value1"}})
 	})
+	t.Run(env, func(t *testing.T) {
```
Comment (gbbr) [inline cmd/trace-agent/config/config_test.go:844]: One empty line above please.

File: cmd/trace-agent/config/config_test.go [hunk 2]
```diff
@@ -853,6 +863,16 @@ func TestLoadEnv(t *testing.T) {
 		assert.NoError(err)
 		assert.Equal(cfg.RejectTags, []*config.Tag{{K: "bad1", V: "value1"}})
 	})
+	t.Run(env, func(t *testing.T) {
```
Comment (gbbr) [inline cmd/trace-agent/config/config_test.go:866]: ditto

File: pkg/config/apm.go [hunk 3]
```diff
@@ -110,10 +110,26 @@ func setupAPM(config Config) {
 	})
 
 	config.SetEnvKeyTransformer("apm_config.filter_tags.require", func(in string) interface{} {
+		if len(in) > 0 && in[0] == '[' {
```
Comment (gbbr) [inline pkg/config/apm.go:113]:
```suggestion
		if len(in) > 0 && in[0] == '[' {
		    // '[' as a first character signals array format
```

File: pkg/config/apm.go [hunk 4]
```diff
@@ -110,10 +110,26 @@ func setupAPM(config Config) {
...
+			var values []string
```
Comment (gbbr) [inline pkg/config/apm.go:114]: I think the code would have less indentation overall if you'd reverse the condition and return with the `strings.Split`. There'd be 1 line indented vs 6.

File: pkg/config/apm.go [hunk 5]
```diff
@@ -110,10 +110,26 @@ func setupAPM(config Config) {
 	})
 
 	config.SetEnvKeyTransformer("apm_config.filter_tags.require", func(in string) interface{} {
```
Comment (gbbr) [inline pkg/config/apm.go:112]: Perhaps you could make a helper function out of it.
```suggestion
	config.SetEnvKeyTransformer("apm_config.filter_tags.require", parseKVList(""apm_config.filter_tags.require""))
```

File: releasenotes/notes/Array-Syntax-for-DD_APM_FILTER_TAGS-f29b167d13cf88d6.yaml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+enhancements:
+  - |
+    DD_APM_FILTER_TAGS_REQUIRE and DD_APM_FILTER_TAGS_REJECT can now be a literal json array.
```
Comment (gbbr) [inline releasenotes/notes/Array-Syntax-for-DD_APM_FILTER_TAGS-f29b167d13cf88d6.yaml:4]:
```suggestion
    APM: DD_APM_FILTER_TAGS_REQUIRE and DD_APM_FILTER_TAGS_REJECT can now be a literal JSON array.
```

File: pkg/config/apm.go [hunk 7]
```diff
@@ -134,6 +130,21 @@ func setupAPM(config Config) {
 	})
 }
 
+func parseKVList(key string) func(string) interface{} {
+	return func(in string) interface{} {
+		if len(in) == 0 || in[0] != '[' {
+			return strings.Split(in, " ")
+		}
```
Comment (gbbr) [inline pkg/config/apm.go:137]: Although clever, I find it a bit confusing. I'd prefer being more verbose here:
```suggestion
		if len(in) == 0 {
			return []string{}
		}
		if in[0] != '[' {
			return strings.Split(in, " ")
		}
```
WDYT?

**Review by gbbr — APPROVED** [review]
Thanks! 👏🏻 🚀

**Pull request merged**